workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/MockConnectionHandler.cs

```diff
@@ -0,0 +1,90 @@
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    public class MockConnectionHandler : IConnectionHandler
+    {
+        private uint updateId;
+
+        private readonly ViewDiff diff = new ViewDiff();
+
+        public void ClearDiff()
+        {
+            diff.Clear();
+        }
+
+        public void CreateEntity(long entityId, EntityTemplate template)
+        {
+            var handler = new EntityTemplateDynamicHandler(template, entityId, diff);
+            DynamicConverter.ForEachComponent(handler);
+        }
+
+        public void ChangeAuthority(long entityId, uint componentId, Authority newAuthority)
+        {
+            diff.SetAuthority(entityId, componentId, newAuthority);
```

Comment:
Worth noting - this still requires the user to update the authority in the right order

Reply:
Yes that is true - we could add more safety checks to this in the future if we find its weird